helm-control

```diff
@@ -66,64 +73,70 @@ static const int32_t reverseSensorSteps = (reverseMinimum - reverseMaximum);
 static const int32_t forwardSensorSteps = (forwardMaximum - forwardMinimum);
 
 // Calculate the nuetral range midpoint
-static const int32_t neutralMidPoint = (reverseMinimum + forwardMinimum)/2;
+static const int32_t neutralMidPoint = (reverseMinimum + forwardMinimum) / 2;
 
 // This is for showing percentages.
 static const int32_t reversePercentSteps = 100 / reverseSensorSteps;
 static const int32_t forwardPercentSteps = 100 / forwardSensorSteps;
 
+uint8_t oldSwitchVal = 1;
 // This is used to know how many samples we have in our rolling average. It's
 // main use is to deal with start-up when we don't yet have a full array.
-int32_t sampleOver       = 0;
 int32_t smoothedPotValue = 0;
 int32_t oldestPotIndex   = 0;  // This is the array index with the oldest value
 int32_t potTotal         = 0;
-int32_t potLoopCount     = 0;
 int32_t potArray[averageOver];
 
+void initializeAverages() {
+  for (int i = 0; i < averageOver; ++i) {
+    potArray[i] = neutralMidPoint;
+  }
+  potTotal = averageOver * neutralMidPoint;
+  oldestPotIndex = 0;
+}
+
```

Comment:
I like the initialization to neutral value, but as soon the Loop() function will started then handle values arrives and in case handle is in forward or reverse position then engine will produce trust immediately.
On the power on the program must send only neutral values to engine until handle is really in neutral position. This prove that owner handle the boat.

Reply:
I tried to stick with optimizing the existing code, not coming up with completely new functionality.